### PR TITLE
Fix sorting of accepted api changes to work on Chrome

### PR DESCRIPTION
**Pull request by tresat**
Chrome doesn't swap elements unless a 1 or -1 is returned from the sort function, other positive or negative values are ignored.

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

File: build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy [hunk 1]
```diff
@@ -70,7 +70,15 @@ class EnrichedReportRenderer extends GroovyReportRenderer {
                         result.acceptedApiChanges.push(correction);
                     });
                     // Sort the array in place by type, then member
-                    result.acceptedApiChanges = result.acceptedApiChanges.sort((a, b) => (a.type +'#' + a.member) > (b.type + '#' + b.member));
+                    // Note that Firefox is fine with a sort function returning any positive or negative number, but Chrome 
+                    // requires 1 or -1 specifically and ignores higher or lower values
+                    result.acceptedApiChanges = result.acceptedApiChanges.sort((a, b) => {
```
Comment (lptr) [inline build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy:75]: 🧐 : no need to set `result.acceptedApiChanges`, `Array.sort()` sorts in-place.

File: build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy [hunk 2]
```diff
@@ -70,7 +70,15 @@ class EnrichedReportRenderer extends GroovyReportRenderer {
...
                     // Sort the array in place by type, then member
-                    result.acceptedApiChanges = result.acceptedApiChanges.sort((a, b) => (a.type +'#' + a.member) > (b.type + '#' + b.member));
+                    // Note that Firefox is fine with a sort function returning any positive or negative number, but Chrome 
+                    // requires 1 or -1 specifically and ignores higher or lower values
+                    result.acceptedApiChanges = result.acceptedApiChanges.sort((a, b) => { 
+                        if ((a.type +'#' + a.member) > (b.type + '#' + b.member)) {
+                            return 1; 
+                        } else {
+                            return -1;
+                        }
```
Comment (lptr) [inline build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy:80]: 🤔 You could also use `new Intl.Collator("ASCII").compare()` here. You can also avoid string concatenation if you do something like this:

```javascript
let result = compare(a.type, b.type);
if (result == 0) {
    result = compare(a.member, b.member);
}
return result;
```

File: build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy (hunk 1 above)
Comment (tresat) [reply build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy:75]: Good catch.  I even commented that it did this, and at some point stopped doing it...

File: build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy (hunk 2 above)
Comment (tresat) [reply build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy:80]: A good idea, but I want to keep the sorting algorithm consistent with that used by `AbstractAcceptedApiChangesMaintenanceTask`.  Which I also need to update now.

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**